dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -418,6 +421,47 @@ public void Test_Secret_Destroy_Method() {
         }
     }
 
+    
+    public void Test_Validate_Descriptor(final AppDescriptorTestCase testCase)
+            throws IOException, DotDataException, DotSecurityException {
+        final AppDescriptorDataGen descriptorDataGen = new AppDescriptorDataGen();
+        descriptorDataGen.name(testCase.name).key(testCase.key).description(testCase.description)
+                .allowExtraParameters(testCase.allowExtraParameters).iconUrl(testCase.iconUrl);
+        for (final Map.Entry<String, ParamDescriptor> entry : testCase.params.entrySet()) {
+            descriptorDataGen.param(entry.getKey(), entry.getValue());
+        }
+        try (final InputStream inputStream = descriptorDataGen.nextPersistedDescriptor()) {
+            final AppsAPI api = APILocator.getAppsAPI();
+            final User admin = TestUserUtils.getAdminUser();
+            AppDescriptor desc = api.createAppDescriptor(inputStream, admin);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'desc' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5358201)